File: code/0b108.cpp
#include <iostream>
using namespace std;

int main() 
{
    int xa,xb,xc,xd,ya,yb,yc,yd;
    bool kt=false;
    cin>>xa>>ya;
    cin>>xb>>yb;
    cin>>xc>>yc;
    cin>>xd>>yd;
    if (xa+xb==xc+xd && ya+yb==yc+yd)
        kt=true;
    if (xa+xc==xb+xd && ya+yc==yb+yd)
        kt=true;
    if (xa+xd==xb+xc && ya+yd==yb+yc)
        kt=true;
    if (kt==true)
        cout<<"YES";
    else cout<<"NO";
    return 0;
}

File: code/0b108.py
1, y1 = [int(x) for x in input().split()]
x2, y2 = [int(x) for x in input().split()]
x3, y3 = [int(x) for x in input().split()]
x4, y4 = [int(x) for x in input().split()]
kt = False 
if (x1 + x2 == x3 + x4) and (y1 + y2 == y3 + y4):
  kt = True
if (x1 + x3 == x2 + x4) and (y1 + y2 == y3 + y4):
  kt = True
if (x1 + x4 == x2 + x3) and (y1 + y4 == y2 + y3):
  kt = True
if kt == True:
  print('YES')
else:
  print('NO')